nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -632,6 +632,19 @@ sub generateXserverConfig {
     }
 }
 
+my $flakeConfig = <<EOF;
+{
+  outputs = { self, nixpkgs }: {
+    nixosConfigurations.nixos = nixpkgs.lib.nixosSystem {
```

Comment:
How about something like this:

```nix
{
  outputs = inputs: {
    nixosConfigurations = builtins.listToAttrs (builtins.map
      (system: { name = system.config.networking.hostName; value = system; })
      [
        (inputs.nixpkgs.lib.nixosSystem {
          system = $system;
          modules = [
            ./configuration.nix
          ];
        })
      ]);
  };
}
```
This way hostname is not hardcoded and it's not required to change it in two places.

Reply:
This pattern requires all configurations to be evaluated. I'd leave it as is, or at least switch it around.

```suggestion
    nixosConfigurations.nixos = nixpkgs.lib.mapAttrs (attrName: sys: nixpkgs.lib.nixosSystem (args // {
      modules = [
        ({ lib, ... }: { system.config.networking.hostName = lib.mkDefault attrName; })
      ] ++ sys.modules;
    }) {
      localhost = [ ./configuration.nix ];
    };
```

This is still reinventing the wheel though and should be deduplicated with the NixOS test framework (after https://github.com/NixOS/nixpkgs/pull/176557). (I guess also NixOps, but I'm not as confident that we can pull it off there and it has other priorities at this time)

Reply:
I think we want to avoid making the first flake.nix a user sees a functional programming heavy nightmare. I think the average person will be able to see that `nixos` is the hostname, especially if we document it. If they don't understand this connection, then they will have other issues, we are only obscuring it and actively hiding it from the user if we try to get rid of it. Ask yourself, what is a new NixOS user going to understand? The function `mapAttrs`, or the string `nixos`?